### PR TITLE
fix(gui): contain panadapter overlay wheel events

### PR DESCRIPTION
**Pull request by rfoust**
## Summary

Prevent wheel and two-finger scroll events inside the panadapter overlay from propagating to the spectrum and tuning the active slice. A shared `SpectrumOverlayWheelGuard` now routes ordinary Display panel content and closed dropdowns to the panel's vertical scroll area, preserves deliberate wheel handling for sliders, open dropdown lists, tables, and other scrollable controls, and consumes otherwise-unhandled wheel input across Band/XVTR, ANT, DAX, Memory, and the main overlay strip.

The Display pane also re-clamps to the panadapter whenever its host resizes. This preserves a real scrollbar range when a pane that originally fit later becomes taller than the available spectrum area, instead of leaving the pane extended off-screen with a zero-range scrollbar. When all Display controls fit, no visible scrolling is expected, but wheel input remains contained by the overlay.

The regression came from the overlay panel event filter delegating wheel events whenever `childAt()` found a child. If that child ignored the event, Qt continued propagation to `SpectrumWidget`. The later Display `QScrollArea` made the leak particularly visible, while closed `QComboBox` controls continued to own the wheel and change selection.

Reproduction:

1. Open the Display pane.
2. Scroll over ordinary panel content or a closed dropdown.
3. Before this change, the event could tune the slice or change the closed dropdown.
4. After this change, the Display pane scrolls when its content overflows; the dropdown and slice frequency remain unchanged even when all content already fits.

Focused regression coverage verifies Display routing, closed and open dropdown ownership, deliberate slider and table scrolling, scroll-limit containment, re-clamping after the host shrinks, non-scrollable panels, the Memory table, the main overlay strip, and no pass-through to a spectrum probe.

## Constitution principle honored

Principle XI — Fixes Are Demonstrated. The event-routing boundary has focused regression coverage and was exercised through the automation bridge against the working-tree patch on macOS, Debian, and Windows. The resize follow-up was additionally reproduced and verified live on macOS by shrinking an open Display pane from a fully fitting layout to an overflowing one.

## Test plan

- [x] Local build passes (`cmake --build build`)
- [x] Behavior verified on a real radio if applicable
- [ ] Existing tests pass (CI)
- [x] Reproduction steps documented if user-reported bug

Additional verification:

- macOS:
  - Native application build passed.
  - Full test suite passed: 164 passed, one intentional skip, zero failures.
  - Display background and closed-dropdown wheel events moved the scrollbar from 0 to 60.
  - The closed dropdown remained `Auto`; the slice remained at 14.1 MHz.
  - With all controls visible, the Display pane correctly had no scroll range.
  - After shrinking the open window, the pane re-clamped from 596 px to the 471 px spectrum height and exposed a 0–125 scroll range; `Reset to Defaults` remained reachable below the System heading.
- Debian VMware:
  - Native application build and focused regression test passed.
  - Display background and closed-dropdown wheel events moved the scrollbar from 0 to 60.
  - The closed dropdown remained `Auto`; the slice remained at 14.096 MHz.
- Windows VMware:
  - Native MSVC/Qt application build and focused regression test passed.
  - Display background and closed-dropdown wheel events moved the scrollbar from 0 to 60.
  - The closed dropdown remained `Auto`; the slice remained at 14.1 MHz.
- On all three platforms, wheel input over the Band button did not tune the slice.
- Transmit automation remained disabled and the radio reported no active transmission throughout runtime verification.
- `git diff --check`, the accessibility scan, and the strict engine-boundary check passed with no blocking findings.

## Checklist

- [x] Commits are signed (`docs/COMMIT-SIGNING.md`)
- [x] No new flat-key `AppSettings` calls — use nested-JSON-under-one-key
      (Principle V)
- [x] Code is clean-room — not decompiled, disassembled, or
      reverse-engineered from a proprietary binary (Principle IV)
- [x] All meter UI uses `MeterSmoother` (AGENTS.md convention) — no meter UI changed
- [x] Documentation updated if user-visible behavior changed — no documentation change was necessary for this bug fix
- [x] Security-sensitive changes reference a GHSA if applicable — not applicable

### CODE REVIEW

File: src/gui/SpectrumOverlayWheelGuard.cpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+void SpectrumOverlayWheelGuard::routeToDisplayScroll(QWheelEvent* wheelEvent)
+{
+    if (!m_displayScrollArea || !m_displayScrollArea->viewport()) {
+        wheelEvent->accept();
+        return;
+    }
+
+    QWidget* scrollArea = m_displayScrollArea->verticalScrollBar();
```
[the remark this answers is not in the document]
Comment (ten9876) [reply src/gui/SpectrumOverlayWheelGuard.cpp]: Fixed in 4693833f — renamed to `scrollBar`, with a note on why the forward targets the scroll bar rather than the viewport.

**Review by ten9876 — APPROVED** [review]
## Review — contain panadapter overlay wheel events

Read both commits against `main`, traced the guard's routing by hand, and validated the runtime behaviour by building the guard TU plus its test locally (now 33 assertions) and probing two edge cases of my own. All three findings are fixed on the branch, plus the naming thread; I also merged `main` (clean, no conflicts).

**Verified sound as written:** event-filter ordering (`guardTree()` installs before the menu's own filter, and the menu filter no longer touches `Wheel`); no recursion in the forward-to-scrollbar path — I probed zero scroll range and a wheel directly over the scrollbar at its limit, and the `m_forwardingDisplayWheel` re-entrancy guard holds; containment is structurally guaranteed because every guarded root is a direct child of `SpectrumWidget`, so even an unguarded descendant's ignored wheel is consumed at the root; the one dynamic-widget site (`setXvtrBands` rebuilding `m_bandPanel` / creating `m_xvtrPanel`) is correctly re-guarded, and I found no other post-construction widget creation; `layoutDisplayPanel()` reproduces the old sizing math exactly; the parent-resize hook can't loop (the panel is a free-floating child, not in a parent layout); `m_displayPanelVisible` is reset on every hide path.

### Fixed

**1. Display pane still clipped off the host bottom.** `constrainedDisplayPanelSize()` clamps the *height* to the host, but `layoutDisplayPanel()` then bottom-anchored the pane to the menu (`panelY = menuBottom - panelH`), so when the panadapter is shorter than the menu (~194 px) `panelY + panelH == menuBottom > hostHeight` and the tail landed below the host — clipped by the parent and unreachable at any scroll offset. That's the exact symptom the second commit set out to fix, surviving in the rewritten function. Added `constrainedDisplayPanelTop()`, which bottom-anchors and then clamps so the whole pane stays inside the host, and wired it into `layoutDisplayPanel()`.

**2. The positioning bug was invisible to a green suite.** The resize coverage only exercised the pure size helper. The position math is now a pure helper too, and the test composes both exactly as `layoutDisplayPanel()` does — the short-host regression case, the bottom-anchored and top-flush cases, and a sweep asserting containment for every host height from 20–900. I confirmed the sweep and the short-host assertion both **fail** against the pre-fix math and pass after, so it's a real regression test rather than a restatement.

**3. Scroll affordance unreachable over most of the pane.** Sliders span columns 1–2 of nearly every Display row, so wheeling to scroll silently retuned Averaging/FPS/Opacity. The deliberate decision: **unlocked** keeps conventional Qt behaviour (the slider owns the wheel, per #570/#1026); **locked** — the project's existing global controls lock (#745), which is precisely the "let me scroll without changing values" opt-in — now yields the wheel for value controls (sliders, spin boxes, closed combos) and routes it to the pane's scroll area. Navigation controls (item views, scroll-area viewports) keep the wheel unconditionally, since a list is scrolled by it and never valued by it.

Worth noting *why* this is decided in the guard rather than left to `GuardedSlider`: those controls already `ignore()` the wheel when locked, but that only reaches the scroll area via Qt's parent propagation for **spontaneous** events. I probed it — a synthetic wheel is not propagated to the parent chain at all — so the behaviour was neither deterministic nor testable. Deciding it at the boundary this guard now owns makes it both, and it's covered by locked/unlocked assertions in the test.

### Also fixed
The `scrollArea` local in `routeToDisplayScroll()` (actually the `QScrollBar`, shadowing the concept of `m_displayScrollArea`) is renamed to `scrollBar`, with a note on why the forward targets the scroll bar rather than the viewport.

### Tests
`spectrum_overlay_wheel_guard_test` grows 20 → 33 assertions, all passing; the full app links. The test target picks up `DragValuePopup.cpp` so the lock coverage exercises the real `GuardedSlider` rather than a stand-in.

Approving.
